### PR TITLE
Storage: UUID regeneration unification

### PR DESCRIPTION
**Pull request by tomponline**
Ceph and LVM do regeneration of filesystem UUIDs differently so this PR aims to unify their approaches taking the best bits of each.

Also partially undoes https://github.com/lxc/lxd/pull/7071 as this was causing filesystem UUIDs to be regenerated when generating a backup.yaml file which is done when taking new snapshots, which ultimately was slowing down taking new snapshots. A fix for this will be followed in a separate PR.

- Adds BTRFS UUID regeneration to Ceph snapshot mounting (borrowed from LVM).
- Adds XFS nouuid mount option rather than UUID regeneration for LVM snapshot mounting (borrowed from Ceph).
- Adds Ceph mount/unmount logging like LVM.
- Fixes BTRFS transaction rollback when UUID regenerating.
- Removes volume snapshot mount when getting snapshot state info.

### CODE REVIEW

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
@stgraber this fixes the snapshot mounting for getting snapshot size, as well as using nouuid option for xfs for lvm snapshot volume mounting. 

If you get a chance to look at this before tomorrow then it can be reviewed now, if not I will also add the ceph unification part next.

File: lxd/instance_snapshot.go [hunk 1]
```diff
@@ -304,8 +305,26 @@ func snapshotPut(d *Daemon, r *http.Request, sc instance.Instance, name string)
 	return operations.OperationResponse(op)
 }
 
-func snapshotGet(sc instance.Instance, name string) response.Response {
-	render, _, err := sc.Render()
+func snapshotGet(s *state.State, snapInst instance.Instance, name string) response.Response {
+	// As this is a relatively expensive operation we only provide it as a render option when querying
+	// snapshot info and not for every call to Render(). It is important that the snapshot not be mounted
+	// here as mounting a snapshot can trigger a very expensive filesystem UUID regeneration, so we rely on
+	// the driver implementation to get the info we are requesting as cheaply as possible.
```
Comment (stgraber) [inline lxd/instance_snapshot.go:312]: This means that a call to `/1.0/instances?recursion=2` will report a size of 0 then?

That's probably fine, at least for now, though then I'd argue that `Size` on a snapshot ought to be marked as `omitempty`.

I also think we probably still want the sizes reported in `/1.0/instances/NAME/snapshots?recursion=1` as that's what we'd use for `lxc info`.

Comment (tomponline) [reply lxd/instance_snapshot.go:312]: @stgraber I'm happy to add it anywhere its needed, I'll define a function that can be used to generate the option into the Render() function. Given it is likely to be a relatively expensive call (especially when used in a list context where its being used for a list of instance each possibly having several snapshots) so I thought I would start conservatively.

I'll add it to those 2 places you suggested.

RE the `omitempty` option, I considered that, but given my understanding of this field is that it should represent the CoW usage of a snapshot, I'd imagine that 0 is valid value for that size (i.e no differences from the original source volume), which is also why I initialised it to -1 so its clear when the value hasn't been set.

Is my understanding correct or have I missed something?

Comment (tomponline) [reply lxd/instance_snapshot.go]: @stgraber have updated those API routes to populate the snapshot size now.

**Pull request merged**